pkg/infra/serverlock/serverlock_test.go

```diff
@@ -56,3 +57,76 @@ func TestServerLock(t *testing.T) {
 		assert.False(t, gotLock)
 	})
 }
+
```

Comment:
If you do take my suggestion above to modify the context, you could write a test here to ensure that the same context must be used to lock and release.